wandb/sdk/wandb_settings.py

```diff
@@ -1666,7 +1666,9 @@ def _infer_run_settings_from_environment(
         program = self.program or _get_program()
         if program is not None:
             program_relpath = self.program_relpath or _get_program_relpath_from_gitrepo(
-                program, _logger=_logger
+                program,
```

Comment:
Here we should condition getting program relpath from gitrepo. How do we do set program relpath when not in a git repo?

Reply:
Also, we may want to reconsider how the entrypoint is set in `job_builder` when the job_source is set to not git...

Reply:
This is going to be pretty gross but I can do it....

Reply:
Example:
```program
if program is not None:
            # Get root if we are in repo, otherwise is None
            repo = GitRepo()
            root = repo.root
            if self.program_relpath:
                program_relpath = self.program_relpath
            elif root and not self.disable_git:
                program_relpath = self.program_relpath or _get_program_relpath_from_gitrepo(
                    program,
                    _logger=_logger,
                )
            else:
                program_relpath = _get_program_relpath(program, None, _logger=_logger)
```